### PR TITLE
DataLinks: Bring back single click links for Stat, Gauge and BarGauge panel

### LINKED ISSUE #26805: Panel Link Usability Regression with new Stat Panel in 7.x
In my Dashboards that I created in version 6.7, I used small SingelStat Panels with Panel Links as "Buttons with Info", basically they show a summary of the Drilldown Dashboard that they link to by changing color.
I found it particularly great that on hover, the cursor would turn into a pointer and show the link title, making it immediately clear to users that _this is a button that leads me somewhere_, even in kiosk mode.
See old singleStat panel:
image

With the new stats panel, panel links and data links do not turn the cursor into a pointer on hover. So there is basically no way for a user to find out something is a link without knowing they can click it.

Panel links are also invisible in kiosk mode, which makes kind of sense since it is called "kiosk" mode, but I really rather used it as "non-editing, distraction-free mode" for users to browse a set of interlinked dashboards in.

Take this Stat panel with a data link AND a panel link in kiosk mode as an example:
image

**What would you like to be added**:
If there is no special hover behaviour for a panel, no data links and only one panel link, please restore the old behaviour.

**Why is this needed**:
How else should users know that they can click stuff?

### PR DESCRIPTION
**Pull request by dprokop**
Test dashboard: http://localhost:3000/d/O1Mi-z8Gz/panel-and-data-links-in-stat-gauge-and-bargauge

Fixes https://github.com/grafana/grafana/issues/26805

Proposed behavior:
1. First panel link is rendered as a viz link if there are no data links
2. If there are data links -> those take precedence over panel links
3. If more than one data link -> context menu is rendered

~**API change:** Panel links supplier is now passed to a panel component.~

TODO:
- [x] Add DataLinksContextMenu tests

### CODE REVIEW

File: packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx [hunk 1]
```diff
@@ -24,11 +27,27 @@ export const DataLinksContextMenu: React.FC<DataLinksContextMenuProps> = ({ chil
...
+    const linkModel = links()[0];
+    return (
+      <a
+        href={linkModel.href}
+        onClick={linkModel.onClick}
+        target={linkModel.target}
+        title={linkModel.title}
+        style={{ display: 'flex' }}
+        aria-label={selectors.components.DataLinksContextMenu.singleLink}
+      >
```
Comment (torkelo) [inline packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx:48]: think rendering anchor will make this tricky to use in the PieChart where we it's inside an svg

**Review by torkelo — COMMENTED** [review]
Awesome job! 

I am really torn on if panel links should be passed to viz and used like a data link like this, It's both nice but also means we have to ways to do the exact same thing. 

More inclined to explore better ways PanelChrome can render panel links in https://github.com/grafana/grafana/pull/29456  (like in a panel footer) 

Screen Shot 2021-03-04 at 18 48 49

File: packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx:48]: but not sure what the alternative is

**Comment by dprokop** [comment]
>I am really torn on if panel links should be passed to viz and used like a data link like this, It's both nice but also means we have to ways to do the exact same thing.

Agree. This PR is to start conversation. I'm all in to revert panel links logic and only support this approach for data links until we explore alternatives for panel links.

**Comment by dprokop** [comment]
@torkelo reverted the panel links- think this should be good to go :)

**Review by torkelo — APPROVED** [review]
Updated the title and add add to changelog label

File: packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/DataLinks/DataLinksContextMenu.tsx:48]: @dprokop any ideas for how we can make this work when inside an svg?

**Pull request merged**